src/repositories/request-repository.ts

```diff
@@ -0,0 +1,80 @@
+import CID from "cids";
+import { EntityRepository, InsertResult, UpdateResult } from "typeorm";
+import { BaseRepository } from 'typeorm-transactional-cls-hooked';
+
+import Context from "../context";
+import Contextual from "../contextual";
+import { Request, RequestUpdateFields } from "../models/request";
+import { RequestStatus } from "../models/request-status";
+import { config } from "node-config-ts";
+
+@EntityRepository(Request)
+export default class RequestRepository extends BaseRepository<Request> implements Contextual {
+
+  setContext(context: Context): void {}
+
+  /**
+   * Create/updates client request
+   * @param request - Request
+   */
+  public async createOrUpdate(request: Request): Promise<Request> {
+    return this.manager.getRepository(Request).save(request);
```

Comment:
I'm not familiar with `typeorm-transactional-cls-hooked` but using a regular TypeORM repository, I think this could be simply `this.save(request)`.

Reply:
I'm explicitly getting the `manager` because of it. I guess this is not an issue, right?